experimental/auth-logic/wrappers/rekor_wrapper.go

```diff
@@ -122,18 +152,107 @@ func verifyRekordLogSignature(rekordEntry *rekord.V001Entry) (*ecdsa.PublicKey,
+}
 
-	data, err := hex.DecodeString(*rekordEntry.RekordObj.Data.Hash.Value)
+// checkInclusionProof forces a check of the inclusion proof. The method
+// (*models.LogEntryAnon).Validate(...) will check the inclusion proof in
+// LogEntryAnon.Verification, but only if it is non-empty. If it is empty
+// it will not error, so this function just throws an error if the verification
+// is empty
+func checkInclusionProof(logEntryAnon *models.LogEntryAnon, registry strfmt.Registry) error {
+	if logEntryAnon.Verification == nil {
+		return fmt.Errorf("logEntryAnon did not have inclusion proof: %v", logEntryAnon)
+	}
+	return logEntryAnon.Validate(strfmt.Default)
+}
+
+// checkEntryPubKeyMatchesExpectedKey compares the public key of the product
+// team in the Rekor log entry to the key of the product team passed as an
+// input to this wrapper. It returns an error if they are not equal
+// (or if valid keys could not be constructed)
+func checkEntryPubKeyMatchesExpectedKey(rekordEntry *rekord.V001Entry, prodTeamKeyBytes []byte) error {
+	logECDSAPubKey, err := pubKeyBytesToECDSA(rekordEntry.RekordObj.Signature.PublicKey.Content)
 	if err != nil {
-		return nil, fmt.Errorf("could not decode hash of data: %v", rekordEntry.RekordObj.Data.Hash.Value)
+		return fmt.Errorf("Invalid product team key in rekor log entry: %v", logECDSAPubKey)
+	}
+	prodTeamECDSAPubKey, err := pubKeyBytesToECDSA(prodTeamKeyBytes)
+	if err != nil {
+		return fmt.Errorf("Invalid product team public key passed as input: %v", logECDSAPubKey)
+	}
+	if !logECDSAPubKey.Equal(prodTeamECDSAPubKey) {
+		return fmt.Errorf("Input product team key does not match rekor log entry key: %v, %v", logECDSAPubKey, prodTeamECDSAPubKey)
 	}
+	return nil
+}
 
-	sig := rekordEntry.RekordObj.Signature.Content
+// EmitStatement returns the unattributed statement for the rekor log wrapper
+func (rlw RekorLogWrapper) EmitStatement() (UnattributedStatement, error) {
+	// Get principal names for the endorsement file and rekor log entry
+	// by using the app name from the endorsement file
+	endorsementAppName, err := GetAppNameFromEndorsement(rlw.endorsementFilePath)
+	if err != nil {
+		return UnattributedStatement{}, fmt.Errorf("could not get app name from endorsement file: %s, %v", rlw.endorsementFilePath, err)
+	}
+	endorsementPrincipal := fmt.Sprintf(`"%s::EndorsementFile"`, SanitizeName(endorsementAppName))
+	logEntryPrincipal := fmt.Sprintf(`"%s::RekorLogEntry"`, SanitizeName(endorsementAppName))
 
-	ok = ecdsa.VerifyASN1(ecdsaKey, data, sig)
-	if !ok {
-		return nil, fmt.Errorf("could not verify ecdsa signature. key:%v, data:%v, sig:%v ", ecdsaKey, data, sig)
+	// Unpack rekord log entry from bytes into go structs
+	logEntryAnon, err := getLogEntryAnonFromBytes(rlw.rekorLogEntryBytes)
+	if err != nil {
+		return UnattributedStatement{}, fmt.Errorf("couldn't parse rekor log entry from bytes: %v, %v", rlw.rekorLogEntryBytes, err)
+	}
+	entryImpl, err := getEntryImplFromAnon(*logEntryAnon)
```

Comment:
I think most of the code from this line to line 237 can be wrapped into a separate function (e.g., `verifyRekorLogEntry`), called here and tested in `rekor_wrapper_test.go`. I believe the lines that generate auth logic statements could all be moved to after the call to `verifyRekorLogEntry`. So `EmitStatement` first verifies everything, and only generates all the statements if the verification passes.